observer/stream.go

```diff
@@ -100,7 +101,20 @@ func (s *Stream) loadBlock(c chan<- *blockatlas.Block, num int64) {
 	// Not strictly correct nor avoids race conditions
 	// But good enough
 	newNum := atomic.AddInt64(&s.blockNumber, 1)
-	err = s.Tracker.SetBlockNumber(s.coin, newNum)
+	if s.coin == coin.BTC {
+		if len(block.Txs) > 0 {
+			s.updateBlockHeight(newNum, num)
+		} else {
+			logger.Error("no of transaction in the BTC block is 0, so skipping and retrying again", s.logParams,
+				logger.Params{"block": num, "coin": s.coin})
+		}
+	} else {
+		s.updateBlockHeight(newNum, num)
+	}
+}
+
+func (s *Stream) updateBlockHeight(newNum int64, num int64) {
+	err := s.Tracker.SetBlockNumber(s.coin, newNum)
```

Comment:
```suggestion
		if s.coin == coin.BTC && len(block.Txs) < 1 {
		logger.Error(err, "Got an empty BTC block - Not persisting block height number", s.logParams, logger.Params{"block": num, "coin": s.coin})
		return
	}
```

Reply:
It looks like too much.

We can simply check if it's BTC and length < 1 and just return right away